### ISSUE TITLE
Single node failure of etcd brings Teleport 3.1.8 down.

### ISSUE THREAD

**Issue by russjones** [gravitational/teleport#2762]
Running Teleport 3.1.8 with etcd 3.3.10.

When connecting to etcd, Teleport connects to the first endpoint int he list of peers and if that node goes down, Teleport goes down with it (no tsh or web login). 

Does this problem still reproduce on 4.0 and 4.1?

**Comment by klizhentas** [comment]
@russjones you have mentioned that you have reproduced the issue? can you provide more details on the repro

**Comment by russjones** [comment]
@klizhentas I didn't reproduce it, it was reported by a customer: https://gravitational.zendesk.com/agent/tickets/779

**Comment by GeckoSplinter** [comment]
Hello guys

Issue impacting : 
Teleport from at least 3.1.8 to teleport 4.0.2 (etcd version : 3.3.13)
Reproduced with :
- teleport 3.1.8
- teleport 3.2.6
- teleport 4.0.2

Step to reproduce:

Teleport configuration
```yaml
teleport:
  nodename: '{{.hostname}}'
  data_dir: /data
  storage:
    type: etcd
    peers:
      - "https://10.20..1:2379"
      - "https://10.20..2:2379"
      - "https://10.20..3:2379"
      - "https://10.20..4:2379"
      - "https://10.20..5:2379"
    tls_cert_file: /data/teleport-auth-client.pem
    tls_key_file: /data/teleport-auth-client-key.pem
    tls_ca_file: /data/ca-teleport-prod.pem
    insecure: false
    prefix: teleport
  auth_servers:
      - 10.20..10:5000
      - 10.20..11:5000
      - 10.20..12:5000
```

ETCD: stoping 1 node on 5 nodes cluster create no issue at all on the cluster availability.

Stoping the first etcd node in the list `10.20..1` will break teleport.
No response through tsh command / no response on web interface.
Restarting the node and teleport is back online right away.

Stoping the second etcd node in the list `10.20..2` have no impact on teleport.

Teleport is using the first etcd peer in its list, and that break the resilience of teleport.

**Comment by klizhentas** [comment]
Just tried to reproduce this, and had no problem with teleport with the first down. What kind of error are you seeing in the logs?

**Comment by klizhentas** [comment]
It turns out the root cause (and the workaround) is described in this issue:

https://github.com/etcd-io/etcd/issues/9949#issuecomment-418292894

To workaround the problem re-issue TLS certs with SANs of all etcd peers avilable /and/or DNS wildcard entry for discovery/reconnects.

This is inconvenient, but the proper fix is to upgrade to 3.4 client, something we are going to do in 4.x major release.
